matrixportal_m4 build errors with arm-none-eabi-gcc 11.2.0

### CircuitPython version

```python
7.0.0-rc.2
```


### Code/REPL

```
[raph@STANVII atmel-samd]$ make -j20 BOARD=matrixportal_m4
```


### Behavior

```
Use make V=1, make V=2 or set BUILD_VERBOSE similarly in your environment to increase build verbosity.
QSTR updated
../../devices/ble_hci/common-hal/_bleio/Adapter.c: In function 'bleio_adapter_gc_collect':
../../devices/ble_hci/common-hal/_bleio/Adapter.c:933:75: error: expression does not compute the number of elements in this array; element type is 'bleio_connection_internal_t', not 'size_t' {aka 'unsigned int'} [-Werror=sizeof-array-div]
  933 |     gc_collect_root((void **)bleio_connections, sizeof(bleio_connections) / sizeof(size_t));
      |                                                                           ^
../../devices/ble_hci/common-hal/_bleio/Adapter.c:933:75: note: add parentheses around the second 'sizeof' to silence this warning
../../devices/ble_hci/common-hal/_bleio/Adapter.c:74:29: note: array 'bleio_connections' declared here
   74 | bleio_connection_internal_t bleio_connections[BLEIO_TOTAL_CONNECTION_COUNT];
      |                             ^~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [../../py/mkrules.mk:74: build-matrixportal_m4/common-hal/_bleio/Adapter.o] Error 1
make: *** Waiting for unfinished jobs....
```

### Description

Using the latest upstream release the build fails.
``` 
arm-none-eabi-gcc --version
arm-none-eabi-gcc (Arch Repository) 11.2.0
```

### Additional information

Building with the officially recommended v10.2.1 works fine.

--- a/ports/atmel-samd/Makefile
+++ b/ports/atmel-samd/Makefile
@@ -197,7 +197,11 @@ CFLAGS += \
 	-DSAM_D5X_E5X -DSAME51
 endif
 
-
+# GCC 11 adds stringop bounds checks that trigger when writing a memory region
+# we know is ok. It's not clear how to give the compiler the info it needs so
+# disable the checks for now.
+# See: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99578
+CFLAGS += -Wno-stringop-overread -Wno-stringop-overflow
 
 LDFLAGS = $(CFLAGS) -nostartfiles -Wl,-nostdlib -Wl,-T,$(GENERATED_LD_FILE) -Wl,-Map=$@.map -Wl,-cref -Wl,-gc-sections -specs=nano.specs
 LDFLAGS += -flto=$(shell $(NPROC))

